contrail-kubernetes-master/hooks/contrail_kubernetes_master_hooks.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+import json
+import sys
+import yaml
+
+from charmhelpers.core.hookenv import (
+    Hooks,
+    UnregisteredHookError,
+    config,
+    log,
+    relation_get,
+    relation_ids,
+    status_set,
+    relation_set,
+    unit_private_ip,
+)
+from charmhelpers.contrib.charmsupport import nrpe
+
+import contrail_kubernetes_master_utils as utils
+import common_utils
+import docker_utils
+
+
+hooks = Hooks()
+config = config()
+
+
+@hooks.hook("install.real")
+def install():
+    status_set('maintenance', 'Installing...')
+
+    docker_utils.install()
+    status_set("blocked", "Missing relation to contrail-controller")
+
+
+@hooks.hook("config-changed")
+def config_changed():
+    update_nrpe_config()
+    docker_utils.config_changed()
+    utils.update_charm_status()
+
+
+@hooks.hook("contrail-controller-relation-joined")
+def contrail_controller_joined():
+    utils.update_kube_manager_data()
+
+
+@hooks.hook("contrail-controller-relation-changed")
+def contrail_controller_changed():
+    utils.update_charm_status()
+
+
+@hooks.hook("contrail-controller-relation-departed")
+def contrail_cotroller_departed():
```

Comment:
there is no same status in update_charm_status. if charm can't work without controller - then update_charm_status should also check this.